### ISSUE TITLE
Tests won't pass on 32-bit platforms

### ISSUE THREAD

**Issue by DavidZemon** [solemnwarning/timespec#2]
Hi! Firstly, thanks for your library. It's quite a bit easier to deal with the timespec struct with your functions.

When our team started using your library, we noticed your tests do not pass on our 32-bit ARM platform. Fixing them was simple once we figured out what was going on: a few of your test cases are using numbers that overflow `tv_nsec`. We've refactored your project structure quite a bit, so a pull request doesn't make a lot of sense, but here's the patch fix

```patch
From d45114a1848221797c6275c7e3ac464c317bcc5e Mon Sep 17 00:00:00 2001
From: David Zemon <david.zemon@redlion.net>
Date: Thu, 25 Jul 2019 09:29:27 -0500
Subject: [PATCH] Don't run invalid tests on 32-bit platforms

---
 test/timespecTests.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/test/timespecTests.c b/test/timespecTests.c
index d33ad51..3923930 100644
--- a/test/timespecTests.c
+++ b/test/timespecTests.c
@@ -326,6 +326,9 @@ int main()
     TEST_NORMALISE(-5,-1000000000, -6,0);
     TEST_NORMALISE(-5,-1500000000, -6,-500000000);
 
+#if (64 <= __WORDSIZE)
+    /* These tests use more than 32-bits for the nanosecond field and are
+       therefore not expected to succeed on 32-bit platforms */
     TEST_NORMALISE(0,2000000000,  2,0);
     TEST_NORMALISE(0,2500000000,  2,500000000);
     TEST_NORMALISE(0,-2000000000, -2,0);
@@ -337,6 +340,7 @@ int main()
 
     TEST_NORMALISE(-1,500000000,  0,-500000000);
     TEST_NORMALISE(-1,499999999,  0,-500000001);
+#endif
 
     if(result > 0)
     {
@@ -347,4 +351,4 @@ int main()
     }
 
     return result;
-}
\ No newline at end of file
+}
-- 
2.10.5
```

Long story short: just gotta wrap the last few `TEST_NORMALISE` test cases with `#if (64 <= __WORDSIZE)`

**Comment by solemnwarning** [comment]
Thanks for pointing that out. I've reduced the test values so they're in range on 32-bit platforms.
